server/storage/storage_impl.go

```diff
@@ -45,18 +46,64 @@ func newEtcdStorage(client *clientv3.Client, rootPath string, opts Options) *Met
 }
 
 func (s *MetaStorageImpl) GetCluster(ctx context.Context, clusterID uint32) (*metapb.Cluster, error) {
-	return nil, nil
+	key := makeClusterKey(clusterID)
+	value, err := s.Get(ctx, key)
+	if err != nil {
+		return nil, errors.Wrapf(err, "get cluster err, key: %v", key)
+	}
+	meta := &metapb.Cluster{}
+	if err = proto.Unmarshal([]byte(value), meta); err != nil {
+		return nil, ErrMetaGetCluster.WithCausef("proto parse err: %v", err)
```

Comment:
- A more accurate error is better, e.g. ErrParseClusterPb.
- No space between `:` and `%v`.
- The message may be better if it reads: `proto parse failed, err:%v`.

Besides this, some other errors in this PR should be refactored too.